File: 20250324/leetcode2255. 统计是给定字符串前缀的字符串数目/main.js
// 给你一个字符串数组 words 和一个字符串 s ，其中 words[i] 和 s 只包含 小写英文字母 。

// 请你返回 words 中是字符串 s 前缀 的 字符串数目 。

// 一个字符串的 前缀 是出现在字符串开头的子字符串。子字符串 是一个字符串中的连续一段字符序列。

/**
 * @param {string[]} words
 * @param {string} s
 * @return {number}
 */
var countPrefixes = function (words, s) {
  const set = new Set();
  for (let i = 0; i < s.length; i++) {
    set.add(s.substring(0, i + 1));
  }
  let res = 0;
  for (const word of words) {
    if (set.has(word)) res++;
  }
  return res;
};

File: 20250324/leetcode2255. 统计是给定字符串前缀的字符串数目/main.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
  int countPrefixes(vector<string> &words, string s) {
    int res = 0;
    int n = s.size();
    for (string word : words) {
      int m = word.size();
      if (m > n) {
        continue;
      }
      bool flag = true;
      for (int i = 0; i < m; i++) {
        if (word[i] != s[i]) {
          flag = false;
          break;
        }
      }
      res += flag;
    }
    return res;
  }
};

int main(void) {
  Solution solution;
  vector<string> words = {"a", "b", "c", "ab", "bc", "abc"};
  string s = "abc";
  cout << solution.countPrefixes(words, s) << endl;
  return 0;
}